pkg/client/purge.go

```diff
@@ -78,14 +80,14 @@ func PurgeProviderData(ctx context.Context, storage Storage, plugin *plugin.Mana
 	return &result, diags
 }
 
-func removeProviderStaleData(ctx context.Context, storage execution.Storage, plugin *plugin.Manager, provider string, lastUpdateTime time.Time, dryRun bool) (map[string]int, int, error) {
+func removeProviderStaleData(ctx context.Context, storage execution.Storage, plugin *plugin.Manager, provider registry.Provider, lastUpdateTime time.Time, dryRun bool) (map[string]int, int, error) {
```

Comment:
`plugin *plugin.Manager` maybe make it `manager *plugin.Manager`?
Could also make an interface `ProviderSchemaGetter` etc? Not sure.